### DISCUSSION TITLE
httpd formula source server (GitHub) requiring authorisation

### DISCUSSION THREAD

**Discussion by jdeepwell** [Homebrew/discussions#4458] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.15
ORIGIN: https://github.com/Homebrew/brew
HEAD: 763c41f006ab9b44261e313b0aaa24a047431b06
Last commit: 3 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 599c95e6f1b64c7eb7323af28aa068318d0a4945
Core tap last commit: 6 weeks ago
Core tap branch: master
Core tap JSON: 27 Apr 09:34 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit arm_blizzard_avalanche
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 13.3.1-arm64
CLT: 14.3.0.0.1.1679647830
Xcode: 14.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Your Xcode (14.2) is outdated.
Please update to Xcode 14.3 (or delete it).
Xcode can be updated from the App Store.
```

### Description of issue

The server used in the "httpd" formula denies access with this error:
  {"errors":[{"code":"UNAUTHORIZED","message":"authentication required"}]}
e.g. for the URL
https://ghcr.io/v2/homebrew/core/httpd/manifests/2.4.57

I also tried to access this URL from another machine on a different network (to make sure it's not an IP issue).

It seems the server has a configuration issue.
Are formulae hosted on multiple servers? Can I somehow switch to a different host for that formula?

**Comment by carlocab** [comment]
The formula source is available on GitHub: https://github.com/Homebrew/homebrew-core/blob/master/Formula/httpd.rb

If you're referring to the bottles, then you should download then using `brew`, which will handle authentication for you.

**Comment by jdeepwell** [comment]
I _was_ downloading with brew. But as it always hang when trying to access anything from ghcr.io, I checked that URL with curl manually, not knowing that it is normal for GitHub to require authentication and that this normally is handled by brew.

It seems it was a transient error (for several hours) – I just tried again and brew ran through the httpd installation fine.

Sorry to bother you!
Thx!
J.
